util/src/main/java/io/kubernetes/client/util/Yaml.java

```diff
@@ -59,6 +58,100 @@ public class Yaml {
 
   static final Logger logger = LoggerFactory.getLogger(Yaml.class);
 
+  /**
+   * Load an API object from a YAML string representation without already knowing the object type and loads and submit any type of resource
+   * V1Pod
+   *
+   * @param content The YAML content
+   * @return An instantiation of the object.
+   * @throws IOException If an error occurs while reading the YAML.
+   */
+  public static Object loadAndSubmitResource(String content) throws Exception {
+    //loading yaml content as unstructured object
+    Object unstructuredObject = Yaml.load(new StringReader(content));
+
+    if (!(unstructuredObject instanceof Map)) {
+      throw new IllegalArgumentException("Invalid YAML");
+    }
+    //typecasting the unstructured object in map
+    Map<String, Object> yamlMap = (Map<String, Object>) unstructuredObject;
+
+    //getting api version and kind
+    String apiVersion = (String) yamlMap.get("apiVersion");
+    String kind = (String) yamlMap.get("kind");
+
+    if (apiVersion == null || kind == null) {
+      throw new IllegalArgumentException("YAML does not contain apiVersion or kind");
+    }
+
+    //creating variable to extract api group
+    String group;
+    String version;
+
+    // If apiVersion contains '/', then it means there is a specified group for it and we need to get that else it is a default group
+    if (apiVersion.contains("/")) {
+      String[] parts = apiVersion.split("/");
+      group = parts[0];
+      version = parts[1];
+    } else {
+      group = "";
+      version = apiVersion;
+    }
+
+    //creating resource plurals from kind
```

Comment:
Consider using the `Discovery` class here instead https://github.com/kubernetes-client/java/blob/f9c82f31b7b6833808bc8880dfc812f18b81851a/util/src/main/java/io/kubernetes/client/Discovery.java#L37